.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,69 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ codeql-test ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ codeql-test ]
+  
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp', 'csharp', 'python' ]
```

Comment:
Is C and objC options too?

Reply:
Yeah, I'm wondering how this would work with the complicated Unity Objective C++ mixed files (.m)

Reply:
C I believe is in option, objC however may be not according to https://codeql.github.com/docs/codeql-overview/supported-languages-and-frameworks/ . However, the code ql for this failed now.